editor/components/block-settings-menu/converters.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { IconButton } from '@wordpress/components';
+import { getUnknownTypeHandlerName, rawHandler, serialize, getBlockContent } from '@wordpress/blocks';
+import { compose } from '@wordpress/element';
+import { withSelect, withDispatch } from '@wordpress/data';
+
+function Converter( { shouldRender, onClick, small, role } ) {
+	if ( ! shouldRender ) {
+		return null;
+	}
+
+	const label = __( 'Convert to Blocks' );
+	return (
+		<IconButton
+			className="editor-block-settings-menu__control"
+			onClick={ onClick }
+			icon="screenoptions"
+			label={ small ? label : undefined }
+			role={ role }
+		>
+			{ ! small && label }
+		</IconButton>
+	);
+}
+
+const UnknownConverter = compose(
+	withSelect( ( select, { uid } ) => {
+		const { canUserUseUnfilteredHTML, getBlock } = select( 'core/editor' );
+		const block = getBlock( uid );
+		return {
+			block,
+			canUserUseUnfilteredHTML: canUserUseUnfilteredHTML(),
+			shouldRender: ( block && block.name === getUnknownTypeHandlerName() ),
+		};
+	} ),
+	withDispatch( ( dispatch, { block, canUserUseUnfilteredHTML } ) => ( {
+		onClick: () => dispatch( 'core/editor' ).replaceBlocks(
+			block.uid,
+			rawHandler( {
+				HTML: serialize( block ),
```

Comment:
I now see:
`serialize` vs `getBlockContent`

Reply:
it might be more difficult to refactor the way I proposed, having two `withDispatch` implementation is also the way to go.